scripts/recon/train_learning_based.py

```diff
@@ -501,5 +467,47 @@ def train_learned(config):
     log.info(f"Results saved in {save}")
 
 
+def reconstruct_save(_idx, config, crop, i, lensed, lensless, psf, test_set, fp):
+    recon = ADMM(psf)
+    recon.set_data(lensless.to(psf.device))
+
+    res = recon.apply(disp_iter=None, plot=False, n_iter=10)
+    res_np = res[0].cpu().numpy()
+    res_np = res_np / res_np.max()
+    lensed_np = lensed[0].cpu().numpy()
+    lensless_np = lensless[0].cpu().numpy()
+    save_image(lensless_np, f"lensless_raw_{_idx}.png")
+    # -- plot lensed and res on top of each other
+    cropped = False
+    if hasattr(test_set, "alignment"):
+        if test_set.alignment is not None:
+            res_np = test_set.extract_roi(res_np, axis=(0, 1))
+        else:
+            res_np, lensed_np = test_set.extract_roi(res_np, lensed=lensed_np, axis=(0, 1))
+
+        cropped = True
+
+    elif config.training.crop_preloss:
+        assert crop is not None
+
+        res_np = res_np[
+            crop["vertical"][0] : crop["vertical"][1],
+            crop["horizontal"][0] : crop["horizontal"][1],
+        ]
+        lensed_np = lensed_np[
+            crop["vertical"][0] : crop["vertical"][1],
+            crop["horizontal"][0] : crop["horizontal"][1],
+        ]
+        cropped = True
+    if cropped and i == 0:
+        log.info(f"Cropped shape :  {res_np.shape}")
+    save_image(res_np, f"lensless_recon_{fp}{_idx}.png")
+    save_image(lensed_np, f"lensed_{fp}{_idx}.png")
```

Comment:
nice function, you can save the lensed image outside the function as it only has to be done once